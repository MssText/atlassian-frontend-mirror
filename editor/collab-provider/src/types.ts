import {
  CollabParticipant,
  CollabEventTelepointerData,
  CollabSendableSelection,
} from '@atlaskit/editor-common/collab';
import { Step } from 'prosemirror-transform';

export type ConnectedPayload = {
  sid: string;
};

export type InitPayload = {
  doc: any;
  version: number;
};

export type ParticipantPayload = {
  sessionId: string;
  userId: string;
  clientId: string;
  timestamp: number;
};

export type TelepointerPayload = ParticipantPayload & {
  selection: CollabSendableSelection;
};

export type StepJson = {
  from?: number;
  to?: number;
  stepType?: string;
  clientId: string;
  userId: string;
};

export type StepsPayload = {
  version: number;
  steps: StepJson[];
};

export type TitlePayload = {
  title: string;
  clientId: string;
};

export type ChannelEvent = {
  connected: ConnectedPayload;
  init: InitPayload;
  'participant:joined': ParticipantPayload;
  'participant:left': ParticipantPayload;
  'participant:telepointer': TelepointerPayload;
  'participant:updated': ParticipantPayload;
  'steps:commit': StepsPayload & { userId: string };
  'steps:added': StepsPayload;
  'title:changed': TitlePayload;
  disconnect: { reason: string };
};

export type CollabEvent = {
  init: Omit<ConnectedPayload, 'sid'>;
  connected: Pick<ConnectedPayload, 'sid'>;
  presence: {
    joined?: CollabParticipant[];
    left?: Pick<CollabParticipant, 'sessionId'>[];
  };
  telepointer: Omit<CollabEventTelepointerData, 'type'>;
  data: {
    json: StepJson[];
    version: number;
    userIds: string[];
  };
  'local-steps': { steps: Step[] };
  'title:changed': TitlePayload;
};

export interface Storage {
  get(key: string): Promise<string>;
  set(key: string, value: string): Promise<void>;
  delete(key: string): Promise<void>;
}

export interface Config {
  url: string;
  documentAri: string;
  userId: string;
  lifecycle?: Lifecycle;
  storage?: Storage;
  createSocket(path: string): Socket;
  getUser?(
    userId: string,
  ): Promise<
    Pick<CollabParticipant, 'avatar' | 'email' | 'name'> & { userId: string }
  >;
}

interface SimpleEventEmitter {
  on(event: string, fn: Function): SimpleEventEmitter;
}

export interface Socket extends SimpleEventEmitter {
  id: string;
  connect(): Socket;
  emit(event: string, ...args: any[]): Socket;
  close(): Socket;
}

export type LifecycleEvents = 'save' | 'restore';
export type EventHandler = () => void;

export interface Lifecycle {
  on(event: LifecycleEvents, handler: EventHandler): void;
}
