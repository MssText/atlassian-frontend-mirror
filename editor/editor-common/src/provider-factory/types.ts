import { EmojiProvider } from '@atlaskit/emoji/types';
import { MentionProvider } from '@atlaskit/mention/types';
import { TaskDecisionProvider } from '@atlaskit/task-decision/types';
import { MediaProvider } from './media-provider';
import { ExtensionProvider } from '../extensions/types';
import { AutoformattingProvider } from './autoformatting-provider';
import { ContextIdentifierProvider } from './context-identifier-provider';
import { ImageUploadProvider } from './image-upload-provider';
import { CollabEditProvider } from '../collab/types';
import { MacroProvider } from './macro-provider';
import { CardProvider } from './card-provider';
import { QuickInsertProvider } from './quick-insert-provider';
import { ProfilecardProvider } from './profile-card-provider';

export interface Providers {
  mediaProvider?: Promise<MediaProvider>;
  emojiProvider?: Promise<EmojiProvider>;
  mentionProvider?: Promise<MentionProvider>;
  extensionProvider?: Promise<ExtensionProvider>;
  autoformattingProvider?: Promise<AutoformattingProvider>;
  taskDecisionProvider?: Promise<TaskDecisionProvider>;
  contextIdentifierProvider?: Promise<ContextIdentifierProvider>;
  imageUploadProvider?: Promise<ImageUploadProvider>;
  collabEditProvider?: Promise<CollabEditProvider>;
  macroProvider?: Promise<MacroProvider>;
  cardProvider?: Promise<CardProvider>;
  quickInsertProvider?: Promise<QuickInsertProvider>;
  profilecardProvider?: Promise<ProfilecardProvider>;

  activityProvider?: Promise<any>; // TODO: Activity AK component looks deprecated any suggestion?
  presenceProvider?: Promise<any>; // TODO: https://product-fabric.atlassian.net/browse/ED-8592
  reactionsStore?: Promise<any>; // TODO: https://product-fabric.atlassian.net/browse/ED-8593
}

export type ProviderName = keyof Providers;
export type ProviderType<T> = T extends keyof Providers
  ? Providers[T]
  : Promise<any>;

export type ProviderHandler<T extends string = any> = (
  name: T,
  provider?: ProviderType<typeof name>,
) => void;
