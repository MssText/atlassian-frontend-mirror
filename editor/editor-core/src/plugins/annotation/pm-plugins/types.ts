import { PortalProviderAPI } from '../../../ui/PortalProvider';
import { Dispatch, EventDispatcher } from '../../../event-dispatcher';
import { AnnotationTypeProviders } from '../types';

export interface InlineCommentPluginOptions {
  dispatch: Dispatch;
  eventDispatcher: EventDispatcher;
  portalProviderAPI: PortalProviderAPI;
  inlineCommentProvider: AnnotationTypeProviders['inlineComment'];
  pollingInterval?: number;
}
