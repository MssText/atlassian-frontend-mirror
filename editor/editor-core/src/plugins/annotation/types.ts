import React from 'react';
import { AnnotationType } from '@atlaskit/adf-schema';
import { ACTIONS } from './pm-plugins/actions';
import { DecorationSet } from 'prosemirror-view';
import { EditorState, SelectionBookmark } from 'prosemirror-state';

export type AnnotationInfo = {
  id: string;
  type: AnnotationType;
};

export enum AnnotationTypes {
  INLINE_COMMENT = 'inlineComment',
}

type AnnotationComponentProps = {
  /**
   * Selected text (can be used when creating a comment)
   */
  textSelection?: string;

  /**
   * DOM element around selected text (for positioning)
   */
  dom?: HTMLElement;
};

export type AnnotationCreateComponentProps = AnnotationComponentProps & {
  /**
   * Creates an annotation mark in the document with the given id.
   */
  onCreate: (id: string) => void;

  /**
   * Indicates that a draft comment was discarded/cancelled
   */
  onClose?: () => void;
};

export type AnnotationViewComponentProps = AnnotationComponentProps & {
  /**
   * Existing annotations where the cursor is placed.
   * These are provided in order, inner-most first.
   */
  annotations: Array<AnnotationInfo>;

  /**
   * Resolves an annotation with the given ID around the selection.
   */
  onResolve: (id: string) => void;

  /**
   * Removes the annotation from the document
   */
  onDelete?: (id: string) => void;
};

export interface AnnotationState<Type, State> {
  annotationType: Type;
  id: string;
  state: State;
}

export interface AnnotationTypeProvider<Type, State> {
  getState: (
    annotationIds: string[],
  ) => Promise<AnnotationState<Type, State>[]>;
  pollingInterval?: number;
}

export type InlineCommentState = { resolved: boolean };
export type AnnotationTypeProviders = {
  inlineComment: AnnotationTypeProvider<
    AnnotationTypes.INLINE_COMMENT,
    InlineCommentState
  >;
};

export interface AnnotationProvider {
  providers?: AnnotationTypeProviders;
  createComponent?: React.ComponentType<AnnotationCreateComponentProps>;
  viewComponent?: React.ComponentType<AnnotationViewComponentProps>;
}

export type InlineCommentMap = { [key: string]: boolean };
export type InlineCommentAction =
  | {
      type: ACTIONS.INLINE_COMMENT_RESOLVE;
      data: { id: string };
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_STATE;
      data: InlineCommentMap;
    }
  | {
      type: ACTIONS.SET_INLINE_COMMENT_DRAFT_STATE;
      data: {
        drafting: boolean;
        editorState: EditorState;
      };
    }
  | {
      type: ACTIONS.INLINE_COMMENT_UPDATE_MOUSE_STATE;
      data: {
        mouseData: {
          x?: number;
          y?: number;
          isSelecting?: boolean;
        };
      };
    };

export type InlineCommentPluginState = {
  annotations: InlineCommentMap;
  mouseData: {
    x: number;
    y: number;
    isSelecting: boolean;
  };
  draftDecorationSet?: DecorationSet;
  bookmark?: SelectionBookmark<any>;
};

const prefix = 'ak-editor-annotation';
export const AnnotationTestIds = {
  prefix,
  floatingComponent: `${prefix}-floating-component`,
  floatingToolbarCreateButton: `${prefix}-toolbar-create-button`,
};

export const DraftDecorationClassName = `${prefix}-draft-selection`;
