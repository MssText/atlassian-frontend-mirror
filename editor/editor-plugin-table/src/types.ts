export interface TablePluginState {
  scaffold?: boolean;
}
