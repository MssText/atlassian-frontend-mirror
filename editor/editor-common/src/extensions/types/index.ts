export type {
  ExtensionParams,
  ExtensionHandler,
  UpdateExtension,
  Extension,
  ExtensionHandlers,
  UpdateContextActions,
  OnSaveCallback,
  TransformBefore,
  TransformAfter,
  // DEPRECATED
  ParametersGetter,
  AsyncParametersGetter,
} from './extension-handler';

export type { ExtensionProvider } from './extension-provider';

export type {
  ExtensionAutoConvertHandler,
  ExtensionComponentProps,
  ExtensionKey,
  ExtensionManifest,
  ExtensionModule,
  ExtensionModuleAction,
  ExtensionModuleActionHandler,
  ExtensionModuleActionObject,
  ExtensionModuleAutoConvert,
  ExtensionModuleNode,
  ExtensionModuleNodes,
  ExtensionModuleType,
  ExtensionModules,
  ExtensionQuickInsertModule,
  ExtensionType,
  CustomFieldResolver,
  UserFieldContextProvider,
  Icon,
  MaybeADFEntity,
  Deserializer,
  Serializer,
} from './extension-manifest';

export type { Parameters } from './extension-parameters';

export type { MenuItem, MenuItemMap } from './utils';

export { isFieldset, isDateRange } from './field-definitions';
export type {
  BooleanField,
  CustomField,
  DateField,
  DateRangeField,
  DateRangeResult,
  EnumField,
  EnumCheckboxField,
  EnumRadioField,
  EnumSelectField,
  FieldDefinition,
  Fieldset,
  NativeField,
  NestedFieldDefinition,
  NumberField,
  Option,
  StringField,
  StringOneLineField,
  StringMultilineField,
  UserField,
  UserFieldContext,
  FieldHandlerLink,
} from './field-definitions';
