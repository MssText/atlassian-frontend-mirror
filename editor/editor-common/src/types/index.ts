import { Node } from 'prosemirror-model';

export interface Transformer<T> {
  encode(node: Node): T;
  parse(content: T): Node;
}

export enum SortOrder {
  ASC = 'asc',
  DESC = 'desc',
  NO_ORDER = 'no_order',
}

export {
  AnnotationState,
  AnnotationProviders,
  AnnotationUpdateEmitter,
  AnnotationUpdateEvent,
  AnnotationUpdateEventPayloads,
  InlineCommentSelectionComponentProps,
} from './annotation';

export { TypeAheadItem, TypeAheadItemRenderProps } from './typeAhead';
