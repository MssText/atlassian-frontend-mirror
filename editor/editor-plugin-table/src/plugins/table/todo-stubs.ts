export const sendLogs = (_a?: any) => {};
