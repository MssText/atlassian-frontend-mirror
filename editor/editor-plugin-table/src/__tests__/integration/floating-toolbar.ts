import {
  animationFrame,
  editable,
  fullpage,
  getDocFromElement,
} from '@atlaskit/editor-test-helpers/integration/helpers';
import {
  clickFirstCell,
  multiCellTableSelectionBottomRightToMiddleTopCell,
  tableSelectors,
} from '@atlaskit/editor-test-helpers/page-objects/table';
import {
  goToEditorTestingWDExample,
  mountEditor,
} from '@atlaskit/editor-test-helpers/testing-example-page';
import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { TableCssClassName } from '../../plugins/table/types';
import basicTable from './__fixtures__/basic-table';
import { documentWithMergedCells } from './__fixtures__/merged-rows-and-cols-document';

BrowserTestCase(
  'should floating toolbar context menu sit above other context menu layers',
  { skip: [] },
  async (client: any) => {
    const page = await goToEditorTestingWDExample(
      client,
      'editor-plugin-table',
    );
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTables: {
        advanced: true,
        allowColumnSorting: true,
        allowDistributeColumns: true,
        allowCellOptionsInFloatingToolbar: true,
      },
      defaultValue: basicTable,
    });
    const tableFloatingToolbarContextMenuSelector = `div[aria-label="Table floating controls"][data-editor-popup=true]`;
    // Select the all cells under column 2 and 3
    await multiCellTableSelectionBottomRightToMiddleTopCell(page);
    // Table floating toolbar should appear, then click on "Cell Options", which brings up another context menu
    const cellOptionsMenuItem = await (
      await page.$(tableFloatingToolbarContextMenuSelector)
    ).$(`button=${tableSelectors.cellOptionsFloatingToolbarText}`);
    await cellOptionsMenuItem.waitForClickable();
    await cellOptionsMenuItem.click();
    // Get the z-index style value of the delete icon popup
    const popupZIndex = await page.evaluate(() => {
      const popupSelector = 'div[aria-label="Popup"][data-editor-popup=true]';
      const deleteIconSelector = 'button.pm-table-controls__delete-button';
      return +(
        (
          document
            ?.querySelector(`${popupSelector} ${deleteIconSelector}`)
            ?.closest(popupSelector) as HTMLElement | null
        )?.style.zIndex || 0
      );
    });
    // Get the z-index style value of the table floating toolbar context menu
    const floatingTablePopupZIndex = await page.evaluate(() => {
      const floatingTablePopupSelector =
        'div[aria-label="Table floating controls"][data-editor-popup=true]';
      return +(
        (
          document?.querySelector(
            floatingTablePopupSelector,
          ) as HTMLElement | null
        )?.style.zIndex || 0
      );
    });
    // Floating toolbar context menu should sit above the popup
    expect(floatingTablePopupZIndex).toBeGreaterThan(popupZIndex);
  },
);

BrowserTestCase(
  'should show hover indicators on delete columns menu option',
  { skip: ['safari'] }, // The test does not pass on CI but works on physical browser
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(
      client,
      'editor-plugin-table',
    );
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTables: {
        advanced: true,
        allowColumnSorting: true,
        allowDistributeColumns: true,
        allowCellOptionsInFloatingToolbar: true,
      },
      defaultValue: basicTable,
    });
    const tableFloatingToolbarContextMenuSelector = `div[aria-label="Table floating controls"][data-editor-popup=true]`;

    // First click on the cell
    await clickFirstCell(page);

    // Table floating toolbar should appear, then click on "Cell Options", which brings up another context menu
    const cellOptionsMenuItem = await (
      await page.$(tableFloatingToolbarContextMenuSelector)
    ).$(`button=${tableSelectors.cellOptionsFloatingToolbarText}`);
    await cellOptionsMenuItem.waitForClickable();
    await cellOptionsMenuItem.click();

    // Hover on the "Delete column" menu option on the context menu
    const deleteColumnMenuItem = await (
      await page.$(tableFloatingToolbarContextMenuSelector)
    ).$(`button=${tableSelectors.removeColumnText}`);
    await deleteColumnMenuItem.moveTo();
    await animationFrame(page);

    // Check the first column cells should have the hover indicators appear
    const hoverCells = await page.$$(
      `tbody tr th.${TableCssClassName.HOVERED_CELL}.${TableCssClassName.HOVERED_COLUMN}, tbody tr td.${TableCssClassName.HOVERED_CELL}.${TableCssClassName.HOVERED_COLUMN}`,
    );
    expect(hoverCells.length).toBe(3);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'should show hover indicators on delete rows menu option',
  { skip: ['safari'] }, // The test does not pass on CI but works on physical browser
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(
      client,
      'editor-plugin-table',
    );
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTables: {
        allowColumnSorting: true,
        allowDistributeColumns: true,
        allowCellOptionsInFloatingToolbar: true,
      },
      defaultValue: basicTable,
    });
    const tableFloatingToolbarContextMenuSelector = `div[aria-label="Table floating controls"][data-editor-popup=true]`;

    // First click on the cell
    await clickFirstCell(page);

    // Table floating toolbar should appear, then click on "Cell Options", which brings up another context menu
    const cellOptionsMenuItem = await (
      await page.$(tableFloatingToolbarContextMenuSelector)
    ).$(`button=${tableSelectors.cellOptionsFloatingToolbarText}`);
    await cellOptionsMenuItem.waitForClickable();
    await cellOptionsMenuItem.click();

    // Hover on the "Delete row" menu option on the context menu
    const deleteRowMenuItem = await (
      await page.$(tableFloatingToolbarContextMenuSelector)
    ).$(`button=${tableSelectors.removeRowText}`);
    await deleteRowMenuItem.moveTo();
    await animationFrame(page);

    // Check the first row cells should have the hover indicators appear (second row from the top)
    const hoverCells = await page.$$(
      `tbody tr:nth-child(2) td.${TableCssClassName.HOVERED_CELL}.${TableCssClassName.HOVERED_ROW}`,
    );
    expect(hoverCells.length).toBe(3);

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'should show tooltip on hover on disabled sort button then remove it on mouse out',
  { skip: ['safari'] }, // The test does not pass on CI but works on physical browser
  async (client: any, testName: string) => {
    const page = await goToEditorTestingWDExample(
      client,
      'editor-plugin-table',
    );
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTables: {
        allowColumnSorting: true,
        allowDistributeColumns: true,
        allowCellOptionsInFloatingToolbar: true,
      },
      defaultValue: documentWithMergedCells,
    });
    const tableFloatingToolbarContextMenuSelector = `div[aria-label="Table floating controls"][data-editor-popup=true]`;
    const sortAtoZButtonSelector =
      'div[data-role=droplistContent] div[role=presentation]';
    const sortZtoAButtonSelector =
      'div[data-role=droplistContent] div[role=presentation]:nth-of-type(2)';
    const tooltipSelector = 'div.atlaskit-portal div[role=tooltip]';

    // Click on the cell on the table
    await clickFirstCell(page);

    // Table floating toolbar should appear, then hover on "Cell Options", which brings up another context menu
    const cellOptionsMenuItem = await (
      await page.$(tableFloatingToolbarContextMenuSelector)
    ).$(`button=${tableSelectors.cellOptionsFloatingToolbarText}`);
    await cellOptionsMenuItem.waitForClickable();
    await cellOptionsMenuItem.click();

    // No tooltip is shown
    expect((await page.$$(tooltipSelector)).length).toBe(0);

    // Hover Sort column A -> Z button then it should show tooltip
    const sortAtoZButton = await page.$(sortAtoZButtonSelector);
    await sortAtoZButton.moveTo();
    const tooltip = await page.$(tooltipSelector);
    await tooltip.waitForExist();
    expect((await page.$$(tooltipSelector)).length).toBe(1);

    // Tooltip should be removed after mouseout to next button
    const sortZtoAButton = await page.$(sortZtoAButtonSelector);
    await sortZtoAButton.moveTo();
    await tooltip.waitUntil(async () => {
      return (await page.$$(tooltipSelector)).length === 0;
    });
  },
);
