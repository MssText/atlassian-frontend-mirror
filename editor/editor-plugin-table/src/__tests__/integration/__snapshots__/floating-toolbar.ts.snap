// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show hover indicators on delete columns menu option 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`should show hover indicators on delete rows menu option 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`should show yellow highlight on the megred rows when hover disabled sort column ASC menu option 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 2,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`should show yellow highlight on the megred rows when hover disabled sort column DESC menu option 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
        "localId": "abc-123",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 2,
                "colwidth": null,
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
